### DISCUSSION TITLE
Any tips on debugging SeaTalk communications issues

### DISCUSSION THREAD

**Discussion by ryanneve** [AK-Homberger/Seatalk-Autopilot-Remote-Control#8] [Q&A]
I've got a new Raymarine ST2000+ tiller pilot and I've been unable to get the remove to move it. There are no other ST devices on the bus. The LED blinks, so I know the remote is working.

I read the closed issue "R3 wrong value" and I think I'm within specifications. I've got 2.9V on pin 3 of the 74LS07 which seems close. I've got 11.9 on the remote's ST when disconnected from the tiller pilot, but it drops when I press a button on the remote as I'd expect it to.

I think I'll haul an oscilloscope and very long extension cord out to the boat next week, but I'm not sure what to look for. Any suggestions!

I'm compiling with Arduino 1.8.13 and have HardwareSerial9bit installed.

**Comment by AK-Homberger** [comment]
Hi,

The first thing to check is that the devive is receiving what it sends. With 12 V connected to the device you should not get any "Send Error" messages on the OLED display.

If you don't have an OLED dispaly pleas add "Serial.println("Send Error);" after "Display("Send Error", 2);" in line 109.
Then you can monitor errors also on the USB-Serial line with the Arduino IDO Serial Monitor.

Regards,
Andreas

**Comment by ryanneve** [comment]
I had the same problem as "R3 wrong value?" I'm using a [TI SN7407N](https://www.ti.com/lit/ds/symlink/sn7407.pdf).

Added Serial.println() to confirm it couldn't receive what it sends.

If anyone else is having this problem, pin 4 on the SN7407N should be low (0.7V) when ST is pulled to ground and 5v when it's near 12v (10.9 on mine when not connected to anything).

I tried changing R3 to a 4.7k, then a 2.7k, but it didn't start working until I used a 2.2k resistor. My "low" value for Pin3/D1 is down to 1.1 volts, which is still a bit above the low level input voltage spec of 0.8v.

Seems to work on the bench, hopefully I'll get it to the boat soon!

**Comment by ryanneve** [comment]
Re-installed it on the boat (ST2000+) and it works in auto mode, but not in standby. Are the ST commands to power-steer different?

**Comment by AK-Homberger** [comment]
Good to know that it works now in auto mode (as it should). Regarding power-steering, I have no idea because I don't own a ST2000+ tiller pilot.

Feel free to try other commands from the list: http://www.thomasknauf.de/rap/seatalk2.htm

**Comment by AK-Homberger** [comment]
const PROGMEM uint16_t ST_Minus_1[] =  { 0x186, 0x21, 0x05, 0xFA };

You can try to use "0x01" or "0x11" instead of "0x21" for all codes in the program (0x11=Z101 remote control or 0x01=ST 1000+).

**Comment by ryanneve** [comment]
I noticed that and was curious what the difference was between x21 and x11. I wonder what the "pressed longer than 1 second" codes do. 
I was also trying to come up with a clever way to do tacking or auto/standby with only four buttons, but no good ideas yet.

**Comment by Fietsenwiel** [comment]
Can you tell me which code works with a Raymarine P70 autopilot with a wheeldrive?
Is that the ST4000?
The originate remote control is a s100 with a baseunit.

I now tried 0x21 but that does not work.

Thanks.

Thanks in advance.

**Reply by AK-Homberger** [reply]
The code worked for me with the p70 and acu 200 without any changes. But you need an Seatalk to SeatalkNG gateway.

**Reply by Fietsenwiel** [reply]
Ok, thanks, I have that.
I see the following when sending:

Datagram +1 is send
Collision
Collision
Collision
Collision
Collision
Send Error
Collision
Collision
Collision
Collision
Collision
Send Error
Beep On, key received
Collision
Collision
Collision
Collision
Collision
Send Error
So something goes wrong. have you any idea?

**Reply by PatrickS1990** [reply]
> Ok, thanks, I have that. I see the following when sending:
> 
> Datagram +1 is send Collision Collision Collision Collision Collision Send Error Collision Collision Collision Collision Collision Send Error Beep On, key received Collision Collision Collision Collision Collision Send Error So something goes wrong. have you any idea?

@Fietsenwiel I have the same problem, were you able to solve it?

**Reply by AK-Homberger** [reply]
Colision Warning means the device can't read what it sends. There are typically two reasons for that behaviour:

1.  9-Bit-Serial patch is not working correctly
2.  Hardware problem. For example critical 7407 devive (try to repace with (74LS07 ore even better 74HC07).

**Reply by PatrickS1990** [reply]
Thank you, I have now ordered the 74HC07. 
Let's see if there will be an improvement.

Edit: 
I have now received and inserted the 74HC07. I have also loaded the code again on the Arduino. Unfortunately still Send Error and Collision.
I now do not know what to do. Does anyone still have an idea?

**Comment by AK-Homberger** [comment]
The hardware is not working properly. It doesn’t receive what it sends.

**Comment by PatrickS1990** [comment]
Hey @ryanneve,
did you get it to work? I would also like to use the remote on an ST2000+.

Greetings
Patrick

**Reply by ryanneve** [reply]
@PatrickS1990 Unfortunately, I haven't been doing the kind of sailing where I use an autopilot, but I hope that changes soon.

**Reply by PatrickS1990** [reply]
@ryanneve Thanks for your message. So nobody has got the remote control working on a ST2000+ yet.

**Reply by ryanneve** [reply]
@PatrickS1990 I worked for me in Auto mode, it just didn't do power steering an standby.

**Reply by PatrickS1990** [reply]
@ryanneve It does not work for me even in Auto mode. What codes did you use?

**Reply by ryanneve** [reply]
@PatrickS1990 , It looks like I didn't make any changes of substance. I was hoping that I would be able to implement two presses at once on the remote, but that wasn't the case. I did purchase a different style remote with many more buttons, but I haven't looked into it much. I think my needs would be better met with hardwired buttons I can position at the front of my cockpit by my jib winches.

**Reply by PatrickS1990** [reply]
@ryanneve Thanks for your information, I still have a problem with collisions and send error. Now I have to see where the fault is.

**Reply by PatrickS1990** [reply]
Still have Collision Errors :(

**Reply by Fietsenwiel** [reply]
me too, I still don't know how to solve it.... My hardware looks ok.

**Reply by PatrickS1990** [reply]
With the help of some electronics guys, I was able to get my remote to work. I used a SN74LS07N from Texas Instruments. I guess it has to be specifically the one from TI for it to work.

**Comment by hirppa10** [comment]
I have also the same problem. I have tried SN74LS07N and Toshiba 74HC07AP. Both gives me send error and collision error. 

EDIT:
I got it working. Third 74LS07 made it work...

**Comment by AK-Homberger** [comment]
That's good news.
It's important to get a "real" 74LS07. But that's sometimes difficult to know before ordering.
